src/main/java/org/radarcns/management/web/rest/OAuthClientsResource.java

```diff
@@ -206,8 +206,10 @@
         try {
             ClientDetails existing = clientDetailsService.loadClientByClientId(clientDetailsDTO
                     .getClientId());
-            throw new CustomConflictException("An OAuth client with that ID already exists",
+            if(Objects.nonNull(existing)) {
```

Comment:
We don't necessarily need that if statement here. If the client does not exist the `NoSuchClientException` will be thrown and this line won't be reached. If the client exists the exception won't be thrown and this line will be reached, and thus throwing the `CustomConflictException`.

Reply:
Indeed. But having an unused local variable is a violation. Stupid way of fixing that violation. Pushed a fix now it.